client/components/editor-standalone/field-definitions/recurring-rules.ts

```diff
@@ -0,0 +1,52 @@
+import {IAuthoringFieldV2, ICommonFieldConfig} from 'superdesk-api';
+import {superdeskApi} from '../../../superdeskApi';
+import {IFieldDefinition} from './interfaces';
+import {cloneDeep, set} from 'lodash';
+import {nameof} from 'core/helpers/typescript-helpers';
+import moment, {Moment} from 'moment';
+
+export const getRecurringRulesField = (): IFieldDefinition => {
+    return {
+        fieldId: 'recurring_rules',
+        getField: ({required, id}) => {
+            const fieldConfig: ICommonFieldConfig = {
+                required: required,
+            };
+
+            const field: IAuthoringFieldV2 = {
+                id: id,
+                name: superdeskApi.localization.gettext('Recurring Rules'),
+                fieldType: 'recurring_rules-react',
+                fieldConfig: fieldConfig,
+            };
+
+            return field;
+        },
+        storageAdapterEvent: {
+            retrieveStoredValue: (item) => {
+                const clonedValue = cloneDeep(item.dates.recurring_rule);
+
+                if (clonedValue?.until != null) {
+                    set(clonedValue, nameof<typeof clonedValue>('until'), moment(clonedValue.until));
```

Comment:
since value update path is static rather than dynamic - no need to use `lodash.set` here. It'd be cleaner to use the usual immutable updating via spreading.